tools/hooks/pre-push

```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
+
+# An example hook script to verify what is about to be pushed.  Called by "git
+# push" after it has checked the remote status, but before anything has been
+# pushed.  If this script exits with a non-zero status nothing will be pushed.
+#
+# This hook is called with the following parameters:
+#
+# $1 -- Name of the remote to which the push is being done
+# $2 -- URL to which the push is being done
+#
+# If pushing without using a named remote those arguments will be equal.
+#
+# Information about the commits which are being pushed is supplied as lines to
+# the standard input in the form:
+#
+#   <local ref> <local sha1> <remote ref> <remote sha1>
+#
+# This script should be copied or better yet sym-linked into .git/hooks/
+
+invoke tox
```

Comment:
Yeah. Agreed. I've been thinking about this since I wrote it. So the thing is, you are completely right about the analysis, we do not need to run the tests on multiple python verisons. The only thing is that tox has the ability to run the checks (test, lint etc) in parallel, which is really great since it speeds things up when the code base grows even a little and you don't ever want these hooks to take time since then they will become an annoyance and people will switch them off.

I mean to fix this by making a special tox task in invoke, that runs the tests only in one python environment. I will update the code and you can have a look.

Reply:
I updated the code and gave the tox task a ``--single`` argument and I changed the hooks to use that. In my view this is the best of both worlds. We will have parrallel execution and independently of what invoke checks does, the pre-push hook will execute the same exact tests that the CI tools will on github, which means that you can't get in that annoying situation where you don't notice that something fails until you already made the PR and then you have to do a small fix-commit. If you want to test it, remember to update your hook if you copied the file rather than soft link it. I will mark the comment resolved because I think that this solves your comment, but re-open if you don't agree.